### PR TITLE
Send images as byte[]

### PR DESCRIPTION
**Pull request by molly-moen**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
All requests were being transformed into json, which caused issues when trying to send images. Any request with an image* content-type will be sent as a byte[]. Also fixed a typo
...

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: ProxyAgent/HttpClient/HttpRequest.cs [hunk 1]
```diff
@@ -147,5 +147,12 @@ public IHttpRequest SetContent<T>(T sourceObject, Encoding encoding, MediaTypeHe
             this.ContentType = mediaType;
             return this;
         }
+
+        public IHttpRequest SetContent(byte[] content, string mediaType)
```
Comment (ppathan) [inline ProxyAgent/HttpClient/HttpRequest.cs]: Why not use one of the generics above SetContent<T>?

Comment (molly-moen) [reply ProxyAgent/HttpClient/HttpRequest.cs]: Those sets the content as a StringContent, and the content should be ByteArrayContent

File: ProxyAgent/HttpClient/HttpRequest.cs [hunk 2]
```diff
@@ -43,6 +43,8 @@ public interface IHttpRequest
         IHttpRequest SetContent<T>(T sourceObject, Encoding encoding, string mediaType);
 
         IHttpRequest SetContent<T>(T sourceObject, Encoding encoding, MediaTypeHeaderValue mediaType);
+
+        IHttpRequest SetContent(byte[] content, string mediaType);
```
Comment (dluc) [inline ProxyAgent/HttpClient/HttpRequest.cs]: could you use `MediaTypeHeaderValue mediaType` here like the previous method?

File: ProxyAgent/Proxy.cs [hunk 3]
```diff
@@ -281,12 +289,19 @@ private string GetRequestPayload(HttpRequest request)
...
                     throw new RequestPayloadTooLargeException();
                 }
             }
 
             return text;
         }
+
+        private byte[] GetByteRequestPayload(HttpRequest request)
+        {
+            var memstream = new MemoryStream();
```
Comment (dluc) [inline ProxyAgent/Proxy.cs]: should we use `using` here to finalize the object?

File: ProxyAgent/Proxy.cs [hunk 4]
```diff
@@ -198,7 +199,14 @@ private IHttpRequest BuildRequest(HttpRequest requestIn, string toHostname)
             var method = requestIn.Method.ToUpperInvariant();
             if (HttpClient.HttpClient.MethodsWithPayload.Contains(method))
             {
-                requestOut.SetContent(this.GetRequestPayload(requestIn));
+                if (requestIn.ContentType.ToLowerInvariant().StartsWith(IMAGE))
```
Comment (dluc) [inline ProxyAgent/Proxy.cs]: You might actually consider to always use binary transfer, that would work right? If I remember correctly, that's what the service does with the response.

The approach here looks a little bit hacky, I would suggest to find a list of common binary mediatypes, make it a list, and check for the value (also use case insensitive comparison). e.g.
* `image/png`
* `image/jpeg`
* `image/gif`
* `application/octet-stream`
* `application/zip`
* `application/pdf`
* `application/msword`
etc.

**Pull request merged**